impeller/display_list/dl_dispatcher.cc

```diff
@@ -994,9 +1001,27 @@ void TextFrameDispatcher::save() {
 
 void TextFrameDispatcher::saveLayer(const DlRect& bounds,
                                     const flutter::SaveLayerOptions options,
-                                    const flutter::DlImageFilter* backdrop) {
+                                    const flutter::DlImageFilter* backdrop,
+                                    std::optional<int64_t> backdrop_id) {
   save();
 
+  if (backdrop != nullptr && backdrop_id.has_value()) {
+    const std::unordered_map<int64_t, BackdropData>::iterator& existing =
```

Comment:
```suggestion
    std::unordered_map<int64_t, BackdropData>::iterator existing =
```

We're grabbing a reference to an rvalue.

Reply:
I might have misunderstood your comment on the double lookup. I tried:

```
    std::unordered_map<int64_t, BackdropData>::iterator existing =
        backdrop_data_.find(backdrop_id.value());
    if (existing == backdrop_data_.end()) {
      // backdrop_data_[backdrop_id.value()] = BackdropData{.backdrop_count = 1, .last_backdrop = backdrop};
      existing->second =
          BackdropData{.backdrop_count = 1, .last_backdrop = backdrop};
    } else {
...
```
But that immediately crashes

Reply:
Oh yea, you've already addressed the double lookup in the branch in the case where it already exists.  When entry doesn't exist you'll have to do a lookup and an insert.  I'm just saying to get rid of the reference to the rvalue here.  You are using a pointer to the iterator that was on the stack and isn't guaranteed to be alive at this point.

Reply:
Oh, I guess technically in C++17 there is variant of `insert` that can take the iterator as a hint.  I've never used that before but could be faster: https://en.cppreference.com/w/cpp/container/unordered_map/insert

Reply:
Done (not std::insert)